server/src/main/java/org/elasticsearch/cluster/metadata/OperatorErrorMetadata.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.metadata;
+
+import org.elasticsearch.cluster.Diff;
+import org.elasticsearch.cluster.SimpleDiffable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * A metadata class to hold error information about errors encountered
+ * while applying a cluster state update for a given namespace.
+ * <p>
+ * This information is held by the {@link OperatorMetadata} class.
+ * </p>
+ */
+public record OperatorErrorMetadata(Long version, ErrorKind errorKind, List<String> errors)
```

Comment:
The error metadata is set when a file fails to get processed for some reason. In this case we launch a cluster state update task that will update this metadata. Example here:

https://github.com/elastic/elasticsearch/pull/86224/files#diff-a7da9bd5e6ce66769b5113745baa42c88b7c21b916a8ea290434048fc5af2c5fR29

The builder code is now removed, so the OperatorErrorMetadata will be directly created.